tests/playwright/README.md

```diff
@@ -5,13 +5,40 @@ This is particularly useful if you want to execute the tests from a Podman Deskt
 - Have a clone of the Podman Desktop repo (you can get it [here](https://github.com/containers/podman-desktop/tree/main))
-- Have pnpm installed (you can get it [here](https://pnpm.io/installation))
+
+## Core functionality how the E2E tests works
+
+Podman Desktop E2E tests uses playwright testing framework libraries to developer the tests and POM (page object model) for the Podman Desktop application. The test runner also relies on playwright code.
+
+At this moment the electron support is marked as [experimental](https://playwright.dev/docs/api/class-electron).
+
+In order to be able to reuse and share the core e2e tests libraries and definitions we created `@podman-desktop/tests-playwright` [npm package](https://www.npmjs.com/package/@podman-desktop/tests-playwright).
+
+The project uses `devDependency` to `@playwright/test` for access to playwright function, assertions, etc. and also to a `electron` package, which brings ability to run Podman Desktop app (which is electron based) when running e2e tests.
+
+### Running E2E tests in development or production mode
+
+We distinguish between two operational modes to run the tests with. Development and production. In both cases, we need to have access to electron binary which is used to [launch the podman desktop app](https://playwright.dev/docs/api/class-electron#electron-launch). Above functionality is handled by `@podman-desktop/tests-playwright` `Runner` class (`podman-desktop-runner.ts`). The runner is then shipped as playwright fixture and takes care of all the setup for the e2e tests.
+
+Multiple scenarios may occur:
+
+1. Running e2e tests from `podman-desktop` repo in DEVELOPMENT mode: We do not need to set any paths to electron binary, `Runner` uses default configuration and grabs electron from `/node_modules/.bin/electron` which is installed via electron dependency postinstall script during `pnpm install`.
+2. Running e2e tests from `podman-desktop` repo in PRODUCTION mode: We need to set `PODMAN_DESKTOP_BINARY` env. var when running the tests from `podman-desktop` repo with the previously built or downloaded Podman Desktop electron app. Simply point to a `podman-desktop` binary or executable.
+3. Running e2e tests in external repository in DEVELOPMENT mode: Set `PODMAN_DESKTOP_ARGS` and point to previous (see later) built `podman-desktop` repo. We are using electron binary from `podman-desktop` repository the same way as in point 1.
+4. Running e2e tests in external repository in PRODUCTION mode: Set `PODMAN_DESKTOP_BINARY` env. var and point to a downloaded/built Podman Desktop App which is also main electron binary that is used.
+
+This approach is beneficial in the way that we do not need to have the `electron` package set as dependency in external repos.
+
+If you set `PODMAN_DESKTOP_BINARY` and `PODMAN_DESKTOP_ARGS` at the same time, runner will throw an error since you cannot really run tests in both mode at the same time.
```

Comment:
```suggestion
If you set `PODMAN_DESKTOP_BINARY` and `PODMAN_DESKTOP_ARGS` at the same time, the runner will throw an error since you cannot really run tests in both mode at the same time.
```